File: app/src/main/java/hr/caellian/notestream/gui/ActivityTagEdit.java
package hr.caellian.notestream.gui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import hr.caellian.notestream.NoteStream;
import hr.caellian.notestream.R;
import hr.caellian.notestream.data.playable.Playable;
import hr.caellian.notestream.lib.Constants;

/**
 * Created by caellyan on 27/06/17.
 */

public class ActivityTagEdit extends AppCompatActivity {
    Playable edited;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tag_edit);

        edited = NoteStream.getInstance().library.savedMusic.getPlayable(getIntent().getStringExtra(Constants.EXTRA_PLAYABLE));
        if (edited == null) {
            edited = NoteStream.getInstance().library.localMusic.getPlayable(getIntent().getStringExtra(Constants.EXTRA_PLAYABLE));
        }

        if (edited == null) {
            Toast.makeText(ActivityTagEdit.this, getString(R.string.edit_unsaved), Toast.LENGTH_LONG).show();
            finish();
        }
    }
}

File: app/src/main/java/hr/caellian/notestream/gui/ActivityTagEdit.kt
/*
 * The MIT License (MIT)
 * NoteStream, android music player and streamer
 * Copyright (c) 2018 Tin Švagelj <tin.svagelj.email@gmail.com> a.k.a. Caellian
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 */

package hr.caellian.notestream.gui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast

import hr.caellian.notestream.NoteStream
import hr.caellian.notestream.R
import hr.caellian.notestream.data.playable.Playable
import hr.caellian.notestream.lib.Constants

class ActivityTagEdit : AppCompatActivity() {
    internal var edited: Playable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tag_edit)

        edited = NoteStream.instance.data.savedMusic.getPlayable(intent.getStringExtra(Constants.EXTRA_PLAYABLE))
        if (edited == null) {
            edited = NoteStream.instance.data.localMusic.getPlayable(intent.getStringExtra(Constants.EXTRA_PLAYABLE))
        }

        if (edited == null) {
            Toast.makeText(this@ActivityTagEdit, getString(R.string.edit_unsaved), Toast.LENGTH_LONG).show()
            finish()
        }
    }
}
